File: project-euler/problem_10/solution.swift
var prime_numbers: [Int] = []
var number: Int = 2


func isPrime(num: Int) -> Bool {
    for i: Int in 2...Int(Double(num).squareRoot() + 1) {
        if num % i == 0 {
            return false
        }
    }

    return true
}


while number < 2000000 {
    if isPrime(num: number) {
        prime_numbers.append(number)
    }

    number += 1
}


print(prime_numbers.reduce(0, +))

File: project-euler/problem_10/solution.dart
void main(List<String> args) {
  List<int> primeNumbers = [];
  int number = 2;

  while (number < 2000000) {
    if (isPrime(number)) {
      primeNumbers.add(number);
    }
    number += 1;
  }

  int sumOfPrimes =
      primeNumbers.fold(0, (previousValue, element) => previousValue + element);
  print(sumOfPrimes);
}

bool isPrime(int number) {
  for (var i = 2; i <= number / i; i++) {
    if (number % i == 0) {
      return false;
    }
  }

  return true;
}
